nemo/collections/llm/recipes/mixtral_8x7b_64k.py

```diff
@@ -1,29 +1,46 @@
-from typing import Callable
-
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Optional
+
+import nemo_run as run
+import pytorch_lightning as pl
 import torch
 
-from nemo.collections.llm.api import pretrain
+from nemo.collections.llm.api import finetune, pretrain
 from nemo.collections.llm.gpt.data.mock import MockDataModule
 from nemo.collections.llm.gpt.data.squad import SquadDataModule
 from nemo.collections.llm.recipes import mixtral_8x7b
-from nemo.collections.llm.utils import Config, Partial
 from nemo.utils.exp_manager import TimingCallback
 
 NAME = "mixtral_8x7b_64k"
 
 
-def pretrain_recipe(
-    name: str, ckpt_dir: str, num_nodes: int, num_gpus_per_node: int, fn: Callable = pretrain
-) -> Partial:
-    recipe = mixtral_8x7b.pretrain_recipe(
-        name=name, ckpt_dir=ckpt_dir, num_nodes=num_nodes, num_gpus_per_node=num_gpus_per_node, fn=fn
-    )
+@run.cli.factory(name=NAME)
+def model() -> run.Config[pl.LightningModule]:
+    model_config = mixtral_8x7b.model()
+    model_config.config.seq_length = 65536
+    model_config.config.max_position_embeddings = 65536
+    return model_config
 
-    model = mixtral_8x7b.model()
-    model.config.seq_length = 65536
-    model.config.max_position_embeddings = 65536
 
-    trainer = mixtral_8x7b.trainer(
+def trainer(
+    num_nodes: int = 1,
```

Comment:
Can we change the default `num_nodes` to TPPPCP/8? For all the long context recipes?